### PR TITLE
Pass proper timeouts when communicating with XDBC bridge.

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Pass proper timeouts when communicating with XDBC bridge. Recently timeouts were not respected when checking bridge liveness and receiving meta info.

This issue was reproduced by Thread Fuzzer. See:

https://clickhouse-test-reports.s3.yandex.net/9814/d50148217fd3df765c039fc167b4995216c926a5/functional_stateless_tests_(release)/test_run.txt.out.log

I'm not sure that it was working with zero timeouts before. Need to carefully read the code. @alesapin ?

### CODE REVIEW

**Comment by alesapin** [comment]
Poco uses non-blocking sockets for `::connect` and I have found the following code https://github.com/ClickHouse-Extras/poco/blob/master/Net/src/SocketImpl.cpp#L133. It uses `pollImpl` to check connection  https://github.com/ClickHouse-Extras/poco/blob/master/Net/src/SocketImpl.cpp#L459-L475 only if connect return non zero code. With zero timeouts, if the connection still not ready it will immediately return zero and throw an exception. So seems like we are just lucky in most cases (because our tests work on localhost).

File: src/Common/XDBCBridgeHelper.h [hunk 1]
```diff
@@ -109,7 +109,8 @@ class XDBCBridgeHelper : public IXDBCBridgeHelper
             uri.setPath(IDENTIFIER_QUOTE_HANDLER);
             uri.addQueryParameter("connection_string", getConnectionString());
 
-            ReadWriteBufferFromHTTP buf(uri, Poco::Net::HTTPRequest::HTTP_POST, nullptr);
+            ReadWriteBufferFromHTTP buf(
+                uri, Poco::Net::HTTPRequest::HTTP_POST, {}, ConnectionTimeouts(http_timeout, http_timeout, http_timeout));
```
Comment (alesapin) [inline src/Common/XDBCBridgeHelper.h:113]: Maybe we should use `ConnectionTimeouts::getHTTPTimeouts(context)`?

Comment (alexey-milovidov) [reply src/Common/XDBCBridgeHelper.h]: Done.

**Comment by alexey-milovidov** [comment]
> ClickHouse build check — 19/20 builds are OK 

No idea what has happened. It looks like another Sandbox failure.

**Pull request merged**